### PR TITLE
feat: add refresh button (#1986)

### LINKED ISSUE #1986: Implement no-code git Refresh button as defined
Is this ready to go after https://github.com/SwissDataScienceCenter/renku-ui/issues/1956 is done?

### PR DESCRIPTION
**Pull request by andre-code**
As decided in #1956, the plan is to handle only the simplest case where the session is behind the orgin, but is not dirty and has not diverged.

1. Session is not behind origin (could be ahead)

image

2. Session is strictly behind origin

image

3. Session is dirty

image

4. Session is behind and ahead of origin

image

6. No sidecar or anonymous

image

Closes #1986 
/deploy renku=000-fix-acceptance-tests-after-changing-gitlab-url extra-values=gateway.gitlabUrl=https://gitlab.dev.renku.ch/ #persist

### CODE REVIEW

**Comment by ciyer** [comment]
As decided in #1956, the plan is to handle only the simplest case where the session is behind the orgin, but is not dirty and has not diverged.

1. Session is not behind origin (could be ahead)

image

2. Session is strictly behind origin

image

3. Session is dirty

image

4. Session is behind and ahead of origin

image


6. No sidecar or anonymous

image

File: client/src/notebooks/components/PullSession.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import { Button, Col, Modal, ModalBody, ModalHeader, Row } from "../../utils/ts-wrappers";
+import { useGitStatusQuery, useHealthQuery, useRenkuPullMutation } from "../../features/session/sidecarApi";
+import type { GitStatusResult } from "../../features/session/sidecarApi";
+
+import { Notebook } from "./Session";
+import { CenteredLoader, InformationalBody, commitsPhrasing } from "./Sidecar";
+import type { CloseModalProps, ModalProps } from "./Sidecar";
+
+interface PullSessionProps extends ModalProps {
+  hasPullAccess: boolean;
```
Comment (andre-code) [inline client/src/notebooks/components/PullSession.tsx:30]: I think in this iteration isLogged and hasPullAccess are not needed. If that is the case could you please remove it.

**Review by andre-code — COMMENTED** [review]
It works very well 👏, also thanks for the refactoring of the components, it looks great.
I have found only 2 small suggestions.

File: e2e/cypress/integration/local/session.spec.ts [hunk 2]
```diff
@@ -72,4 +72,81 @@ describe("display a session", () => {
...
+    cy.gui_open_session();
+    // save session
+    cy.get_cy("save-session-button").click();
+    cy.get(".modal-dialog").should("exist");
+    cy.get(".modal-dialog").get("h5").contains("Save Session").should("be.visible");
+    cy.get(".modal-dialog").get("div")
+      .contains("You have work in this session that has not yet been saved to the server.").should("be.visible");
+  });
+
+  it.only("pull changes button -- no sidecar", () => {
```
[the remark this answers is not in the document]
Comment (ciyer) [reply e2e/cypress/integration/local/session.spec.ts:109]: Oops!

**Comment by ciyer** [comment]
> While the session is starting, the buttons are available and return the same message as for anonymous sessions. I would either disable them with a popover saying "Session isn't ready", or leave them enabled and add a similar text to the bigger modal !

Good point! This is exactly what was done in #2103.

File: client/src/notebooks/components/PullSession.tsx (hunk 1 above)
Comment (ciyer) [reply client/src/notebooks/components/PullSession.tsx]: Good point, done!

**Review by andre-code — COMMENTED** [review]
It works as expected and all changes were implemented.
Thanks 🙌

🚀

**Review by ciyer — APPROVED** [review]

**Pull request merged**